packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -355,26 +351,68 @@ export function CreateTaskForm({
 
   const isScreenHeightLessThan800 = useMediaQuery('(max-height:800px)');
 
-  const [favoriteTaskBuffer, setFavoriteTaskBuffer] = React.useState<TaskFavorite>(
-    defaultFavoriteTask(),
-  );
+  // Note that we are not checking if the number of supported tasks is larger
+  // than 0, this will cause the dashboard to fail when the create task form is
+  // opened. This is intentional as it is a misconfiguration and will require
+  // the build-time configuration to be fixed.
+  let allTasksToDisplayAreValid = true;
+  const validTasks: TaskDefinition[] = [];
+  let defaultTaskDescription: string | TaskDescription | null = null;
+  let defaultTaskRequest: TaskRequest | null = null;
+  tasksToDisplay.forEach((supportedTask: TaskDefinition, index: number) => {
```

Comment:
Should memo this

Reply:
https://github.com/open-rmf/rmf-web/pull/925/commits/f2c0b08afc76c006185537d6fc4e44f12c441cb7